tests/utils.py

```diff
@@ -0,0 +1,30 @@
+import time
+from typing import Any, Callable
+
+import snap_http
+from snap_http.types import SnapdResponse
+
+
+def call_and_await_api(name: str, *args: Any, **kwargs: Any) -> SnapdResponse:
+    """Call the `name` endpoint and wait until changes are applied."""
+    f: Callable[..., SnapdResponse] = getattr(snap_http, name)
+    response = f(*args, **kwargs)
+
+    if response.type == "sync":
+        return response
+
+    final_states = {"Done", "Abort", "Hold", "Error"}
```

Comment:
style nit: consider moving to global/constant set or enum. Not critical.

Reply:
Moved to a global set